admin/kernel-masterdata-service/src/test/java/io/mosip/kernel/masterdata/test/service/MasterDataServiceTest.java

```diff
@@ -3128,4 +3082,91 @@ public void getImmediateChildrenByLocCodeTestDataExceptionTest() {
 		locationHierarchyService.getImmediateChildrenByLocCode("KAR", List.of("eng"));
 	}
 
+	@Test
+	public void buildZoneFilter_withZoneList_returnSearchFilters() {
+		List<Zone> zones = new ArrayList<>();
+		Zone zone1 = new Zone();
+		zone1.setCode("java");
+		zone1.setLangCode("eng");
+		zone1.setHierarchyPath("hierarchy");
+		zone1.setHierarchyLevel((short)1);
+		zone1.setParentZoneCode("Parent");
+		zone1.setName("name");
+		zone1.setHierarchyName("name");
+		zone1.setIsActive(true);
+		zones.add(zone1);
+
+		List<SearchFilter> filters = (List<SearchFilter>) ReflectionTestUtils.invokeMethod(registrationCenterService, "buildZoneFilter", zones);
+		assertNotNull(filters);
+		assertEquals(zones.size(), filters.size());
+		ReflectionTestUtils.invokeMethod(registrationCenterService, "buildZoneFilter", zones);
+	}
+
+	@Test
+	public void buildZoneFilter_withZoneCode_returnSearchFilters() {
+		String zoneCode = "zone";
+		SearchFilter filter = new SearchFilter();
+		filter.setColumnName(MasterDataConstant.ZONE_CODE);
+		filter.setType(FilterTypeEnum.EQUALS.name());
+		filter.setValue(zoneCode);
+		ReflectionTestUtils.invokeMethod(registrationCenterService, "buildZoneFilter", zoneCode);
```

Comment:
There is no assertion nor exception that is caught in the test method.

Reply:
Added assertions